src/OpenTK.Mathematics/Geometry/Ray3.cs

```diff
@@ -0,0 +1,143 @@
+﻿//
+// Ray.cs
+//
+// Copyright (C) 2021 OpenTK
+//
+// This software may be modified and distributed under the terms
+// of the MIT license. See the LICENSE file for details.
+//
+
+
+using System;
+using System.Diagnostics.Contracts;
+
+namespace OpenTK.Mathematics.Geometry {
+
+    /// <summary>
+    /// Defines a ray in 3d space.
+    /// </summary>
+    public struct Ray3 {
+
+        /// <summary>
+        /// Starting point of the ray.
+        /// </summary>
+        public Vector3 Origin { get; set; }
+
+        private Vector3 _direction { get; set; }
```

Comment:
These should be public fields. This is also what all of the other math types do.